src/pages/api/auth/[...nextauth].js

```diff
@@ -52,14 +52,40 @@ export const authOptions = {
             ...response.message,
           };
         } else {
-          return null;
+          return response;
         }
       },
       credentials: {
         email: { label: "Email", type: "text" },
         password: { label: "Password", type: "password" },
       },
     }),
+    CredentialsProvider({
+      id: "signup",
+      name: "Signup with Email and Password",
+
+      async authorize(credentials) {
+        const response = await updateUser(
+          credentials.email,
+          credentials.password,
+          credentials.name,
+        );
+
+        if (response.status === 200) {
+          return {
+            id: response.message._id,
+            ...response.message,
+          };
+        } else {
+          return response;
```

Comment:
this error doesn't appear on the frontend see https://stackoverflow.com/questions/76750792/nextauth-how-to-return-custom-errors-without-redirection